File: result/translate/starchat/java2cpp/c++/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.cpp
string f_gold ( string str, int k ) {
        vector<char> palin(str.begin(), str.end());
        string ans = "";
        int l = 0, r = str.length() - 1;
        while (l < r) {
            if (str[l]!= str[r]) {
                palin[l] = palin[r] = max(str[l], str[r]);
                k--;
            }
            l++, r--;
        }
        if (k < 0) return "Not possible";
        l = 0, r = str.length() - 1;
        while (l <= r) {
            if (l == r) {
                if (k > 0) palin[l] = '9';
            }
            if (palin[l] < '9') {
                if (k >= 2 && palin[l] == str[l] && palin[r] == str[r]) {
                    k -= 2;
                    palin[l] = palin[r] = '9';
                }
                else if (k >= 1 && (palin[l]!= str[l] || palin[r]!= str[r])) {
                    k--;
                    palin[l] = palin[r] = '9';
                }
            }
            l++, r--;
        }
        for (int i = 0; i < palin.size(); i++) ans += palin[i];
        return ans;
    }

File: result/translate/starchat/python2java/java/MAKE_LARGEST_PALINDROME_CHANGING_K_DIGITS.java
public static String f_gold ( String str, int k ) {
        char [] palin = str.toCharArray();
        int l = 0, r = palin.length - 1;
        while ( l <= r ) {
            if ( palin [ l ]!= palin [ r ] ) {
                palin [ l ] = palin [ r ] = Math.max ( palin [ l ], palin [ r ] );
                k--;
            }
            l++;
            r--;
        }
        if ( k < 0 ) return "Not possible";
        l = 0, r = palin.length - 1;
        while ( l <= r ) {
            if ( l == r ) {
                if ( k > 0 ) palin [ l ] = '9';
            }
            if ( palin [ l ] < '9' ) {
                if ( k >= 2 && palin [ l ] == str.charAt ( l ) && palin [ r ] == str.charAt ( r ) ) {
                    k--;
                    palin [ l ] = palin [ r ] = '9';
                }
                else if ( k >= 1 && ( palin [ l ]!= str.charAt ( l ) || palin [ r ]!= str.charAt ( r ) ) ) {
                    k--;
                    palin [ l ] = palin [ r ] = '9';
                }
            }
            l++;
            r--;
        }
        return new String ( palin );
    }